fv/wireguard_test.go

```diff
@@ -348,6 +347,73 @@ var _ = infrastructure.DatastoreDescribe("WireGuard-Supported", []apiconfig.Data
 			Expect(rcvd[1]).NotTo(BeEmpty())
 			Expect(rcvd[1]).To(Equal(sent[0]))
 		})
+
+		for _, ai := range []bool{true, false} {
+			allInterfaces := ai
+			desc := "should add wireguard port as a failsafe"
+			if ai {
+				desc += " (using * HostEndpoint)"
+			} else {
+				desc += " (using eth0 HostEnpoint"
+			}
+			It(desc, func() {
+				By("Creating policy to deny wireguard port on main felix host endpoint")
+				policy := api.NewGlobalNetworkPolicy()
+				policy.Name = "deny-wg-port"
+				prot := numorstring.ProtocolFromString(numorstring.ProtocolUDP)
+				policy.Spec.Egress = []api.Rule{
+					{
+						Action:   api.Deny,
+						Protocol: &prot,
+						Destination: api.EntityRule{
+							Selector: "has(host-endpoint)",
+							Ports:    []numorstring.Port{numorstring.SinglePort(wireguardListeningPortDefault)},
+						},
+					},
+					{Action: api.Allow},
+				}
+				policy.Spec.Ingress = []api.Rule{
+					{
+						Action:   api.Deny,
+						Protocol: &prot,
+						Destination: api.EntityRule{
+							Selector: "has(host-endpoint)",
+							Ports:    []numorstring.Port{numorstring.SinglePort(wireguardListeningPortDefault)},
+						},
+					},
+					{Action: api.Allow},
+				}
+				policy.Spec.Selector = "all()"
+				policy.Spec.Types = []api.PolicyType{api.PolicyTypeIngress, api.PolicyTypeEgress}
+				_, err := client.GlobalNetworkPolicies().Create(utils.Ctx, policy, utils.NoOptions)
+				Expect(err).NotTo(HaveOccurred())
+
+				By("Creating a HostEndpoint for each Felix")
+				for _, f := range felixes {
+					hep := api.NewHostEndpoint()
+					hep.Name = "hep-" + f.Name
+					hep.Labels = map[string]string{
+						"name":          hep.Name,
+						"hostname":      f.Hostname,
+						"host-endpoint": "true",
+					}
+					hep.Spec.Node = f.Hostname
+					hep.Spec.ExpectedIPs = []string{f.IP}
+					if allInterfaces {
+						hep.Spec.InterfaceName = "*"
+					} else {
+						hep.Spec.InterfaceName = "eth0"
+					}
+					_, err := client.HostEndpoints().Create(utils.Ctx, hep, options.SetOptions{})
+					Expect(err).NotTo(HaveOccurred())
+				}
+
+				By("Checking there is connectivity between the workloads")
+				cc.ExpectSome(wls[0], wls[1])
```

Comment:
Could do with some traffic that gets blocked here too, otherwise the check might sneak in before the policy is applied.

Reply:
That's a good idea.  Will do!

Reply:
I've added a Consistently check for now - not ideal, but I think passable